delivery/routes.py

```diff
@@ -22,12 +22,12 @@ def __init__(self, restaurants: AbstractRestaurantRepository, schema):
         self.restaurants = restaurants
         self.schema = schema
 
-    def get(self, restaurant_id: int):
+    def get(self, restaurant_id: str):
         return jsonify(self.schema.dump(self.restaurants.get_by_id(restaurant_id))), 200
 
     def put(self, restaurant_id: int):
         content = request.json
         restaurant = self.schema.load(content)
-        restaurant.id = restaurant_id
+        restaurant._id = restaurant_id
         self.restaurants.update(restaurant)
         return jsonify(self.schema.dump(self.restaurants.get_by_id(restaurant_id))), 200
```

Comment:
so here 2 DB calles heppen:
1. on `restaurants.update`
2. on `restaurants.get_by_id`
DB calls imply calling remote server and network transmission may take quite long.

it is common for `create` and `update` methods to return the resulting object.
please do the following:
1. change the signature of `AbstractRestaurantRepository` to return the resulting (updated) object on `update`, `create`. e.g. 
   ```python
   @abstractmethod
   def create(item: Restaurant) -> Restaurant:
       ...
   ```
2, update repository implementation accordingly 
3. use more efficient calls to Mongo. for example `insert_one` returns the created object, the same goes for [update_one](https://pymongo.readthedocs.io/en/stable/api/pymongo/collection.html#pymongo.collection.Collection.update_one)

this will allow to reduce the number of DB calls and execute the code faster

Reply:
fixed